openshift/osd-network-verifier
# OSD-20513: Add support for the `--noproxy` directive in `curl`

## What does this PR do? / Related Issues / Jira

This adds an additional `Proxy` config to support passing `--noproxy` to `curl` when probing urls. The values passed will _not_ be proxied through the configured http/https proxy.

I opted to not update the legacy probe based on it being called "legacy" but am open to go down that route if we still have use cases for it.

I tested this with the CLI locally and ran the integration tests as well. Local execution of the `no_proxy` behavior via environment variable:

```
> export http_proxy=squid.corp.redhat.com:3128 https_proxy=squid.corp.redhat.com:3128 no_proxy=google.com
> curl redhat.com -w "%{json}" -Z -s -o /dev/null | jq .proxy_used
1
> curl google.com -w "%{json}" -Z -s -o /dev/null | jq .proxy_used
0
```

On the integration tests, I ran into an aws sdk issue that required bumping the dependencies.

## Checklist




## Reviewer's Checklist


- [ ] (This needs to be done after technical review) I've run the branch on my local, verified that the functionality is ok

## How to test this PR locally / Special Instructions



We do not have the integration test setup with an http/https proxy at this time, but you can modify the integration to pass a `Proxy.NoProxy` slice and validate the tests run as expected. We could invest in including a proxy if we want a more easy test.

## Logs
